apps/rush-lib/src/logic/PackageChangeAnalyzer.ts

```diff
@@ -129,10 +129,20 @@ export class PackageChangeAnalyzer {
     //  });
     // }
 
+    // Determine the current variant from the link JSON.
```

Comment:
>// Determine the current variant from the link JSON. [](start = 4, length = 52)

People sometimes do this:

```
rush unlink
(run some NPM commands)
rush link
```

or this:

`rush link --force` (after manually tampering with symlinks)

With the current design, this would cause the saved variant to be lost.  Would it be better to track the variant in its own data file,  e.g. **common/temp/variant.json** or somesuch?

Reply:
We chatted about this and are thinking maybe to use a different file **current-variant.flag**.

---
In reply to: [226876205](https://github.com/Microsoft/web-build-tools/pull/722#discussion_r226876205) [](ancestors = 226876205)

Reply:
And then you would add **current-variant.flag** to the `potentiallyChangedFiles` array in InstallManager.ts.

---
In reply to: [227155249](https://github.com/Microsoft/web-build-tools/pull/722#discussion_r227155249) [](ancestors = 227155249,226876205)

Reply:
I added `current-variant.json`. However, `potentiallyChangedFiles` is checked too late in the process, so instead I check to see if the file is updated at the time it gets written before installation starts. That seems to work well in testing.